File: 27-Interpreter/src/go/variable.go
package main

// Variable is a Terminal Expression — it looks up its value in the context.
type Variable struct {
	name string
}

func NewVariable(name string) *Variable {
	return &Variable{name: name}
}

func (v *Variable) Interpret(context *Context) bool {
	return context.Get(v.name)
}

File: 27-Interpreter/src/rust/variable.rs
use crate::expression::Expression;
use crate::context::Context;

/// Terminal Expression — a variable that looks up its value in the context.
pub struct Variable {
    name: String,
}

impl Variable {
    pub fn new(name: &str) -> Self {
        Variable { name: name.to_string() }
    }
}

impl Expression for Variable {
    fn interpret(&self, context: &Context) -> bool {
        context.get(&self.name)
    }
}
